core/src/main/java/com/googlecode/psiprobe/tokenizer/Tokenizer.java

```diff
@@ -267,18 +267,17 @@ private int lookupSymbol(char chr) throws IOException {
     if (index >= 0) {
       // the index could be anywhere within a group of sybols with the same first letter
       // so we need to scroll up the group to make sure we start test from the beginning
-      while (index > 0 && ((TokenizerSymbol) symbols.get(index - 1)).compareTo(chrObj) == 0) {
+      while (index > 0 && symbols.get(index - 1).compareTo(chrObj) == 0) {
         index--;
       }
       while (index < symbols.size()) {
-        TokenizerSymbol symbol = ((TokenizerSymbol) symbols.get(index));
+        TokenizerSymbol symbol = (symbols.get(index));
```

Comment:
That sounds great!  Last I checked on sonar, there was a pay for use version out there.  It's probably been a year.  I think it was rather cost prohibitive at the time.  I did just quickly look at this.  Sounds like this might be a cheap alternative.  If sonar is accessible, it can be tied into Pull Requests as another check before merging.